### PR TITLE
Test

### PR DESCRIPTION
**Pull request by nickdesaulniers**
very sloppy, will need more work.  Just want to see if it still hangs on travis or not.

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
so the test failures are interesting. basically the specifics of the error message from nn.Err.  I get the two different ones based on whether I run `npm t` vs `make check`.  I wonder what the difference is, and whether it matters?

**Comment by nickdesaulniers** [comment]
actually, the one version of node that passed has a curios failed assertion

File: test/chan.js [hunk 1]
```diff
@@ -196,49 +197,50 @@ test('multi-topic registration followed by calls to rmchan on all but one stops
...
+        sub.rmchan('foo')
+      } else if(msgs === 2) {
+        t.equal(msg, 'bar world');
+        sub.rmchan('bar');
+      } else if (msgs === 3) {
+        t.equal(msg, 'hi world');
+        sub.rmchan('hi');
+      } else if (msgs === 4) {
+        t.equal(msg, 'hello world');
+      } else if (msgs === 5) {
```
Comment (reqshark) [inline test/chan.js]: lets make that one a `switch(msg) { case n: // ...`

**Comment by nickdesaulniers** [comment]
tests are hung, I don't having connect be async makes a difference.

**Comment by reqshark** [comment]
@nickdesaulniers, The async worker threads are helpful but we're skirting around the real issue here, which is [endpoint identifiers](http://nanomsg.org/v0.6/nn_connect.3.html), see e.g. RETURN VALUE 

And while we're at it let's just quickly do a sanity check on your earlier question re: executing `npm i` vs. `make`, which still just executes `npm i` 

> I wonder what the difference is, and whether it matters?

the difference between those operations is zero.

we'll need to deal with the fact nanomsg connect() calls over TCP give us an *endpoint identifier*, which is not the same as an established connection state. Similarities around the ordering observed between those two states is coincidental, (although we really need a lot of latency to experience actual msg loss).

**Comment by nickdesaulniers** [comment]
> we'll need to deal with the fact nanomsg connect() calls over TCP give us an endpoint identifier, which is not the same as an established connection state. Similarities around the ordering observed between those two states is coincidental

so when using TCP as the transport, calling `nn_connect` is not a guarantee of a connection?

**Comment by reqshark** [comment]
correctamundo

**Comment by reqshark** [comment]
well, no i mean eventually it is going to connect.. it has a nice backoff and retry algorithm if it gets really congested or takes a really long time for the bind address to come up..  basically one of the greatest features of nanomsg IMO.. who cares when clients or servers connect, disconnect, reconnect :palm_tree:

**Pull request closed without merging**